mechanics/micromechanics/include/micromechanics/environment.h

```diff
@@ -1,20 +1,71 @@
 #pragma once
 
+#include <array>
+#include <memory>
+
 #include <common/timestep_executor.h>
 #include <common/types.h>
 
+#include "agent_container.h"
+#include "cell_data.h"
+#include "simulation_parameters.h"
+#include "spatial_index.h"
+
 namespace physicore::mechanics::micromechanics {
 
+class solver;
+class vtk_mechanics_serializer;
+
+/**
+ * @brief Main environment class for micromechanics simulations.
+ *
+ * Holds all simulation state: agents, parameters, solver, and spatial index.
+ * The solver is obtained from the solver_registry based on params.solver_name.
+ */
 class environment : public timestep_executor
 {
+public:
+	/// Mechanics timestep
 	real_t timestep;
 
-public:
-	explicit environment(real_t timestep) : timestep(timestep) {}
+	/// Agent container with all agent data
+	std::unique_ptr<agent_container> agents;
 
-	void run_single_timestep() override;
+	/// Cell-level data (pressure, etc.) aggregated from agents
+	cell_data cells;
+
+	/// Simulation parameters including type-based interactions
+	simulation_parameters params;
+
+	/// The active solver (obtained from solver_registry)
+	std::unique_ptr<solver> active_solver;
```

Comment:
maybe just solver is fine. calling it active_solver suggests that there is a time when solver is inactive which is not the case